webpack/components/AnsibleRolesAndVariables/AnsibleRolesAndVariables.js

```diff
@@ -0,0 +1,223 @@
+import React, { useState } from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  cellWidth,
+} from '@patternfly/react-table';
+import { Button,
+  Toolbar,
+  ToolbarGroup,
+  ToolbarContent,
+  ToolbarItem, Checkbox,
+  EmptyStateIcon,
+  Title,
+  EmptyStateBody,
+  EmptyState,
+  Pagination,
+  } from '@patternfly/react-core';
+import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon';
+
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { post } from 'foremanReact/redux/API';
+import { useDispatch } from 'react-redux';
+import { push } from 'connected-react-router';
+import {
+  ANSIBLE_ROLES_INDEX,
+  ANSIBLE_ROLE_CONFIRM_IMPORT_PATH,
+  DEFAULT_PER_PAGE,
+} from './AnsibleRolesAndVariablesConstants';
+import { prepareResult } from './AnsibleRolesAndVariablesHelpers';
+
+const ImportRolesAndVariablesTable = ({
+  columnsData,
+  rowsData,
+  proxy,
+}) => {
+  const columns = columnsData.map(col => ({
+    ...col,
+    transforms: [cellWidth(10)],
+  }));
+  const [rows, setRows] = useState(rowsData);
+
+  const [isChecked, setIsChecked] = useState(false);
+  const [selectedRowsCount, setSelectRowsCount] = useState(0);
+
+  const dispatch = useDispatch();
```

Comment:
redux wiring could be moved to a wrapper component and actions into a separate file, [a rough sketch](https://github.com/xprazak2/foreman_ansible/commit/d0b08e03f501b1dccae2a6c6fb47956bfa84021e)